// sdk-loading/esm2022/lib/sdk-loading.module.mjs
import { NgModule, NO_ERRORS_SCHEMA } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SDKLoadingComponent } from './sdk-loading.component';
import * as i0 from "@angular/core";
export class SDKLoadingModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.1", ngImport: i0, type: SDKLoadingModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.3.1", ngImport: i0, type: SDKLoadingModule, declarations: [SDKLoadingComponent], imports: [CommonModule], exports: [SDKLoadingComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.3.1", ngImport: i0, type: SDKLoadingModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.1", ngImport: i0, type: SDKLoadingModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        SDKLoadingComponent
                    ],
                    imports: [
                        CommonModule
                    ],
                    exports: [
                        SDKLoadingComponent
                    ],
                    schemas: [
                        NO_ERRORS_SCHEMA
                    ]
                }]
        }] });
