// sdk-google-map/esm2022/lib/sdk-google-map.module.mjs
import { NgModule } from '@angular/core';
import { SDKGoogleMapComponent } from './sdk-google-map.component';
import { SDKLoadingModule } from 'sdk-loading';
import * as i0 from "@angular/core";
export class SDKGoogleMapModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.1", ngImport: i0, type: SDKGoogleMapModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.3.1", ngImport: i0, type: SDKGoogleMapModule, declarations: [SDKGoogleMapComponent], imports: [SDKLoadingModule], exports: [SDKGoogleMapComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.3.1", ngImport: i0, type: SDKGoogleMapModule, imports: [SDKLoadingModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.1", ngImport: i0, type: SDKGoogleMapModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        SDKGoogleMapComponent
                    ],
                    imports: [
                        SDKLoadingModule,
                    ],
                    exports: [
                        SDKGoogleMapComponent
                    ]
                }]
        }] });
